test/api/core_test.dart

```diff
@@ -308,6 +321,41 @@ void main() {
       check(st.toString()).contains("distinctivelyNamedFromJson");
     }
   });
+
+  group('ApiConnection user-agent', () {
+    Future<void> checkUserAgent(String expectedUserAgent) async {
+      return FakeApiConnection.with_(account: eg.selfAccount,
+        useBinding: true,
+        (connection) async {
+          connection.prepare(json: {});
+          await connection.get(kExampleRouteName, (json) => json, 'example/route', null);
```

Comment:
Let's also pick any one of the test cases below, and do one check with each of the different request methods: `get`, `post`, `postFileFromStream`, `delete`.